client/src/components/EditProfile/EditForm/EditForm.tsx

```diff
@@ -0,0 +1,140 @@
+import { Formik, Form } from 'formik';
+import * as Yup from 'yup';
+import { Box, Grid, Typography } from '@material-ui/core';
+
+import Textfield from '../../FormsUI/Textfield';
+import Select from '../../FormsUI/Select';
+import DateTimePicker from '../../FormsUI/DateTimePicker';
+import Button from '../../FormsUI/Button';
+import useStyles from './useStyles';
+
+const initial_form_state = {
+  firstName: '',
+  lastName: '',
+  gender: '',
+  birthDate: '',
+  email: '',
+  phone: '',
+  address: '',
+  message: '',
+};
+
+const form_validation = Yup.object().shape({
+  firstName: Yup.string().required('This field cannot be empty'),
+  lastName: Yup.string().required('This field cannot be empty'),
+  gender: Yup.string().required('This field cannot be empty'),
+  birthDate: Yup.date().required('This field cannot be empty'),
+  email: Yup.string().email('Invalid email.').required('This field cannot be empty'),
+  phone: Yup.number().integer().typeError('Please enter a valid phone number').required('This field cannot be empty'),
+  address: Yup.string().required('Required'),
+  message: Yup.string(),
+});
+
+const EditForm = (): JSX.Element => {
+  const classes = useStyles();
+  const options = ['Male', 'Female', 'Other'];
+
+  return (
+    <Box>
+      <Formik
+        initialValues={{
+          ...initial_form_state,
+        }}
+        validationSchema={form_validation}
+        onSubmit={(values) => {
+          console.log(values);
+        }}
+      >
+        <Form>
+          <Grid container alignItems="center" spacing={2}>
+            <Grid item xs={12}>
+              <Typography variant="h6" align="center">
+                Edit Profile
+              </Typography>
+            </Grid>
+            <Grid item xs={12} sm={3}>
+              <Typography variant="body1" className={classes.typography}>
+                First Name
+              </Typography>
+            </Grid>
+            <Grid item xs={12} sm={8}>
+              <Textfield name="firstName" label={<Typography className={classes.label}>firstName</Typography>} />
+            </Grid>
```

Comment:
I'm confused as to why we provide 2 different labels. Shouldn't we use either-or? Also, we might want to provide a `component="label"` prop for correct semantics 👍🏼

Reply:
> 

@ugur-akin Kindly explain more please

Reply:
As Rajiv explained, the initial `Typography` component as well as the `label` prop both have the same purpose - in fact, I'm surprised as to why we don't see 2 labels on the UI here (based on the screenshot).

By spec, [HTML labels](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) are for captions of particular UI elements and best practices compel us to have one for input elements. They are often used with [`for` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/for) ([`htmlFor` in React](https://reactjs.org/docs/dom-elements.html#htmlfor)) to form an association programatically. The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) demonstrates the benefits quite well 👍🏼